### PR TITLE
Unable to use Edge agent with Portainer exposed over node port

### LINKED ISSUE #25: Unable to use Edge agent with Portainer exposed over node port
It is currently not possible to use the Edge compute feature of Portainer when deploying Portainer exposed over node port (works fine when exposed via load balancer).

The following manifest is invalid https://github.com/portainer/k8s/blob/master/deploy/manifests/portainer/portainer.yaml

I'll open a PR to actually fix this manifest but we'll need to check the helm manifest template as well to make sure it is not impacted.

### PR DESCRIPTION
**Pull request by funkypenguin**
Fixes #25

### CODE REVIEW

File: .ci/scripts/generate-portainer-manifests-from-chart.sh [hunk 1]
```diff
@@ -15,7 +15,7 @@
 # 2. Remove the header produced by helf --dry-run
 # 3. Remove references to helm in rendered manifests (no point attaching a label like "app.kubernetes.io/managed-by: Helm" if we are not!)
 
-helm install --no-hooks --namespace zorgburger --set disableTest=true --dry-run zorgburger charts/portainer \
+helm install --no-hooks --namespace zorgburger --set service.type=NodePort --set disableTest=true --dry-run zorgburger charts/portainer \
```
Comment (deviantony) [inline .ci/scripts/generate-portainer-manifests-from-chart.sh:18]: What is this change about?

File: charts/portainer/templates/service.yaml [hunk 2]
```diff
@@ -22,8 +22,13 @@ spec:
       {{- if (and (eq .Values.service.type "NodePort") (not (empty .Values.service.httpNodePort))) }}
       nodePort: {{ .Values.service.httpNodePort}}
       {{- end }}
+      {{- if (eq .Values.service.type "NodePort") }}
```
Comment (deviantony) [inline charts/portainer/templates/service.yaml:25]: I'm a bit confused about the condition here, do we still need to differentiate the `.Values.service.edgeNodePort` and `.Values.service.edgePort` variables?

**Review by deviantony — COMMENTED** [review]
Just a few comments/questions

File: .ci/scripts/generate-portainer-manifests-from-chart.sh (hunk 1 above)
Comment (funkypenguin) [reply .ci/scripts/generate-portainer-manifests-from-chart.sh:18]: The script runs the helm templating process twice - the first one generates nodeport config (but this was previously implicit), and the second generates loadbalancer config. This change simply make nodeport on the first run more explicit, for readability.

File: charts/portainer/templates/service.yaml (hunk 2 above)
Comment (funkypenguin) [reply charts/portainer/templates/service.yaml:25]: If there are still cases where we'd want edgePort to be 8000 (like in a LB deploy), then yes :)

Comment (deviantony) [reply charts/portainer/templates/service.yaml:25]: Indeed.

**Review by deviantony — APPROVED** [review]
LGTM then @funkypenguin you can merge it.

**Pull request merged**